File: project-code/cs50x/problem-set-1/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height;

    // Prompt the user for a valid height between 1 and 8
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // Print the pyramid
    for (int i = 1; i <= height; i++)
    {
        // Print spaces for left pyramid
        for (int j = 0; j < height - i; j++)
        {
            printf(" ");
        }
        // Print hashes for left pyramid
        for (int k = 0; k < i; k++)
        {
            printf("#");
        }
        // Print gap between pyramids
        printf("  ");
        // Print hashes for right pyramid
        for (int k = 0; k < i; k++)
        {
            printf("#");
        }
        // Move to the next line
        printf("\n");
    }
    return 0;
}

File: cs50x/problem-set-6/sentimental-mario-less/mario.py
from cs50 import get_int


def main():
    # Prompt user for the height of the pyramid
    height = 0
    while height < 1 or height > 8:
        height = get_int("Height: ")

    # Print the half-pyramid aligned to the right
    for i in range(1, height + 1):
        # Calculate the number of spaces and hashes for the current row
        spaces = height - i
        hashes = i
        print(" " * spaces + "#" * hashes)


if __name__ == "__main__":
    main()
